enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,641 @@
+1. User associates some BareMetalAssets with the ClusterDeployment.
+1. Hive creates BareMetalHost resources corresponding to each BareMetalAsset, and uses them to boot the live ISO.
+1. baremetal-operator uses redfish virtualmedia to boot the live ISO.
+1. Agent running on each host contacts assisted service, resulting in an AssistedInstallAgent resource being created.
+1. Assisted Installer walks each agent through discovery and validation workflows.
+1. When each agent is in a ready state, hive or ACM initiates cluster creation with the AssistedInstallCluster.
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For example, consider both security and how this will impact the larger OKD
+ecosystem.
+
+How will security be reviewed and by whom? How will UX be reviewed and by whom?
+
+Consider including folks that also work outside your immediate sub-project.
+
+## Design Details
+
+### Open Questions
+
+#### Can we use sqlite?
+
+It would be advantageous to use sqlite for the database. The [gorm library
+supports sqlite](https://gorm.io/docs/connecting_to_the_database.html#SQLite),
+but it is not clear if assisted service is compatible. In particular, the [use
+of FOR
+UPDATE](https://github.com/openshift/assisted-service/blob/e70af7dcf59763ee6c697fb409887f00ab5540f5/pkg/transaction/transaction.go#L8)
+might be problematic.
+
+#### baremetal-operator watching multiple namespaces?
+
+When utilizing baremetal-operator on a hub cluster to boot the discovery ISO on
+hosts, should we be creating those BareMetalHost resources in a separate
+namespace from those that are associated with the hub cluster itself?
+
+What work is involved in having BMO watch additional namespaces?
+
+#### BareMetalAsset: does it have value in these scenarios?
+
+BareMetalAsset is a CRD that is part of ACM. It represents an invetory of
+hardware that ACM can use to provision with IPI and add BareMetalHosts to spoke
+clusters. ACM does not use BareMetalAsset to interact with or manage hardware;
+it only uses it to store hardware information that gets either passed to
+`openshift-install` or added directly to a spoke cluster in the form of a
+BareMetalHost.
+
+Scenario A: user maintains an inventory of hosts as BareMetalAssets, all in one
+namespace, or grouped by organization. At install time, a BareMetalHost is
+created (in the namespace appropriate to the cluster) for each selected
+BareMetalAsset, and then the "Create Cluster" scenario takes place.
+BareMetalHost resources get garbage collected once provisioning is complete.
+
+Scenario B: user maintains an inventory of hosts as BareMetalHosts. They are
+directly used to boot live ISOs as part of agent-based provisioning. When a
+user selects BMHs for cluster creation, they may be moved into a new
+cluter-specific namespace by deleting and re-creating them.
```

Comment:
Deleting and recreating host resources is relatively expensive, because it requires updating ironic, which include ensuring the host is deprovisioned. It seems more flexible to not care which namespace a host is in, so that users can configure their inventory how they prefer.